frontend/js/src/stats/UserListeningActivity.tsx

```diff
@@ -335,22 +335,31 @@ export default class UserListeningActivity extends React.Component<
     const { dateFormat } = this.rangeMap.all_time;
     let totalListens = 0;
     let totalYears = 0;
-
     const allTimeData = [];
     const currYear = new Date().getFullYear();
     for (let i = 2002; i <= currYear; i += 1) {
       const yearData = data.payload.listening_activity.filter(
         (year) => year.time_range === String(i)
       )[0];
-
       totalYears += 1;
       if (yearData) {
         const date = new Date(yearData.from_ts * 1000);
-        allTimeData.push({
-          id: date.toLocaleString("en-us", dateFormat),
-          thisRangeCount: yearData.listen_count,
-          thisRangeTs: yearData.from_ts,
-        });
+        if (yearData.listen_count > 0) {
```

Comment:
One suggestion, don't remove all years with 0 listen count but only the ones before any listening activity is encountered. Say a user started submitting data in 2017, didn't submit anything in 2018 but then started again in 2019. The graph should start from 2017 and 2018 should be shown in the graph. With the current checks, 2018 would be hidden.

Reply:
Hey, any update on this?